Data/IntMap/Internal.hs

```diff
@@ -1740,25 +2433,38 @@ split k t =
 splitLookup :: Key -> IntMap a -> (IntMap a, Maybe a, IntMap a)
```

Comment:
Might it pay to hold GHC's hand using `StrictTriple` for the recursion here? That should make it super-easy for GHC (and us) to see that the three pieces are all in WHNF at each stage and at the end.